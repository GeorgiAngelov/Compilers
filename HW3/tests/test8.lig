function main(args : [[int]]) : int {
      var i : int;
      var b : bool;

      if (i) { }
      if (b) { }
      if (i) { } else { }
      if (b) { } else { }
      while (i) { }
      while (b) { }
      for (i = b to i) { }
      for (i = i to b) { }
      for (i = i to i) { }
      return;
      return (b);
      return (i);
}
