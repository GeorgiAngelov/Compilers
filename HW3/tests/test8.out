(decls
   (decl-fun:NONE (id main) (type-fun (params (typed-id (id args) (type-array (type-array (type-int))))) (type-int))
      (decls
         (decl-var:ok (id i) (type-int))
         (decl-var:ok (id b) (type-bool)))
      (stmts
         (if:NONE (id:int i)
            (stmts)
            (stmts))
         (if:ok (id:bool b)
            (stmts)
            (stmts))
         (if:NONE (id:int i)
            (stmts)
            (stmts))
         (if:ok (id:bool b)
            (stmts)
            (stmts))
         (while:NONE (id:int i)
            (stmts))
         (while:ok (id:bool b)
            (stmts))
         (for:NONE (id:int i) (id:bool b) (id:int i)
            (stmts))
         (for:NONE (id:int i) (id:int i) (id:bool b)
            (stmts))
         (for:ok (id:int i) (id:int i) (id:int i)
            (stmts))
         (return:NONE)
         (return:NONE (id:bool b))
         (return:ok (id:int i)))))
Typechecking failed!
